lib/matplotlib/collections.py

```diff
@@ -172,7 +172,10 @@ def __init__(self, *,
         self._face_is_mapped = None
         self._edge_is_mapped = None
         self._mapped_colors = None  # calculated in update_scalarmappable
-        self._hatch_color = mcolors.to_rgba(mpl.rcParams['hatch.color'])
+        self._hatch_color = mpl.rcParams['hatch.color']
+        if self._hatch_color == 'inherit':
+            self._hatch_color = mpl.rcParams['patch.edgecolor']
+        self._hatch_color = mcolors.to_rgba(self._hatch_color)
```

Comment:
Better to do this with a local variable and only set to the attribute once:

```suggestion
        hatch_color = mpl.rcParams['hatch.color']
        hatch_color == 'inherit':
            hatch_color = mpl.rcParams['patch.edgecolor']
        self._hatch_color = mcolors.to_rgba(hatch_color)
```

it has the advantages of:
 a) being a bit clearer when you read line 175 that this is an intermediary state
 b) `self._hatch_color` is never set to an invalid value (`"inherited"`).  Yes this is in the `__init__`, a private attribute, and it should "always" be fixed by the following lines, it is easy for these details to drift apart over time and it is better to never go to an invalid state